include/zephyr/bluetooth/bluetooth.h

```diff
@@ -908,9 +926,15 @@ struct bt_le_per_adv_param {
 						  _peer)
 
 
-#define BT_LE_ADV_CONN BT_LE_ADV_PARAM(BT_LE_ADV_OPT_CONNECTABLE, \
-				       BT_GAP_ADV_FAST_INT_MIN_2, \
-				       BT_GAP_ADV_FAST_INT_MAX_2, NULL)
+/**
+ * @deprecated It is now mandatory to use @ref BT_LE_ADV_OPT_ONE_TIME
+ * for connectable advertising. Please use @ref BT_LE_ADV_CONN_ONE_TIME
+ * instead.
+ */
+#define BT_LE_ADV_CONN                                                                             \
+	BT_LE_ADV_PARAM(BT_LE_ADV_OPT_CONNECTABLE, BT_GAP_ADV_FAST_INT_MIN_2,                      \
+			BT_GAP_ADV_FAST_INT_MAX_2, NULL)                                           \
+	__DEPRECATED_MACRO
```

Comment:
So what is the plan here? 
Are we 
A) Deprecating this macro and enforcing the use of `BT_LE_ADV_CONN_ONE_TIME` forever, or
B) Deprecating this macro, enforcing `BT_LE_ADV_CONN_ONE_TIME`, and then deprecate `BT_LE_ADV_CONN_ONE_TIME` later again? 

I assume A, just wanted to make sure.

Reply:
This has not been decided. Both scenarios start with this PR as it is.

Reply:
I too would like to avoid breaking our users twice. :+1: Though I'm not too worried about simple renames and other stuff that can be automated by a search-replace or a migration script.

I'm for solution A, but it's actually for different reason: I'm not too worried about the above if the tool-chain or automated tests yell at you if you forget. So failing to compile when there is a semantic difference is desired.

Reusing `BT_LE_ADV_CONN` *ever* introduces the hazard of old code compiling just fine, and even running apparently just fine at first, except eventually discovering that reconnect does not work.

And the symptom of error in this case is just that nothing happens, far removed from the ideal of a self-describing actionable error message. This kind of very late feedback makes our users less confident in upgrading.

I realize now that removing the runtime warning ever will have the same downside. Maybe we should go for deprecating `BT_LE_ADV_OPT_CONNECTABLE`, like I said in https://github.com/zephyrproject-rtos/zephyr/pull/73395#discussion_r1617302003?

We could argue that our users have to "execute" the migration guides one-by-one. :wink: And the lack of API guarantees and verification from Zephyr means any serious application developer should trust-but-verify-tm and have automated test for their applications for all use-cases.

This is becoming an essay. Sorry for the long. I'll table this for BT WG discussion on Thursday.

Reply:
You make some good points. It's arguably better that they get compile errors, rather than some hard-to-track issue if it still compiles and runs. 

In that regard it is better to completely remove the macro in the future.

Reply:
I'd def be in favor of just removing the macro and deprecating the `CONNECTABLE` option in favor of `ONE_TIME`.